src/ResourceManager/Network/Commands.Network/ExpressRouteCrossConnection/CrossConnectionPeering/AzureRMExpressRouteCrossConnectionPeeringBase.cs

```diff
@@ -0,0 +1,137 @@
+    using System.Management.Automation;
+    using MNM = Microsoft.Azure.Management.Network.Models;
+    using System.Linq;
+
+    public class AzureRMExpressRouteCrossConnectionPeeringBase : NetworkBaseCmdlet
+    {
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The name of the Peering")]
+        [ValidateNotNullOrEmpty]
+        public virtual string Name { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The PeeringType")]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet(
+           MNM.ExpressRoutePeeringType.AzurePrivatePeering,
+           MNM.ExpressRoutePeeringType.AzurePublicPeering,
+           MNM.ExpressRoutePeeringType.MicrosoftPeering,
+           IgnoreCase = true)]
+        public string PeeringType { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The PeerAsn")]
+        [ValidateNotNullOrEmpty]
+        public uint PeerASN { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The PrimaryPeerAddressPrefix")]
+        [ValidateNotNullOrEmpty]
+        public string PrimaryPeerAddressPrefix { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The SecondaryPeerAddressPrefix")]
+        [ValidateNotNullOrEmpty]
+        public string SecondaryPeerAddressPrefix { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The vlanId")]
+        [ValidateNotNullOrEmpty]
+        public int VlanId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The SharedKey")]
+        [ValidateNotNullOrEmpty]
+        public string SharedKey { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The MircosoftConfigAdvertisedPublicPrefixes")]
+        [ValidateNotNullOrEmpty]
+        public List<string> MicrosoftConfigAdvertisedPublicPrefixes { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The customerAsn")]
+        [ValidateNotNullOrEmpty]
+        public int MicrosoftConfigCustomerAsn { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "The MircosoftConfigRoutingRegistryName")]
+        [ValidateNotNullOrEmpty]
+        public string MicrosoftConfigRoutingRegistryName { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "PeerAddressType")]
+        [ValidateSet(
+           IPv4,
+           IPv6,
+           IgnoreCase = true)]
+        public string PeerAddressType { get; set; }
+
+        public void ConstructMicrosoftConfig(PSExpressRouteCrossConnectionPeering peering)
+        {
+            if (this.MicrosoftConfigAdvertisedPublicPrefixes != null && this.MicrosoftConfigAdvertisedPublicPrefixes.Any())
```

Comment:
`this.` does nothing in any of these statements (including all the ones below). It is not needed and can be safely removed.

Reply:
Done